### ISSUE TITLE
Debugging fails if Microsoft Azure storage emulator is not running

### ISSUE THREAD

**Issue by BeigeBadger** [MicrosoftDocs/azure-dev-docs#104]
The default `local.settings.json` that is generated looks like this:

```
{
  "IsEncrypted": false,
  "Values": {
    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
    "FUNCTIONS_WORKER_RUNTIME": "python"
  }
}
```

However, if a user tries to follow the guide and start debugging, they will be presented with the following prompt:

![https://i.imgur.com/j0iNzwl.png](https://i.imgur.com/j0iNzwl.png)

If you click "Debug anyway" you will get runtime errors related to the storage:

> "The listener for function 'Functions.ExampleTimerFunction' was unable to start. Microsoft.WindowsAzure.Storage: No connection could be made because the target machine actively refused it. System.Net.Http: No connection could be made because the target machine actively refused it. System.Private.CoreLib: No connection could be made because the target machine actively refused it."

The fix for this is to:

1. Install the Microsoft Azure storage emulator from [here](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator#get-the-storage-emulator) if it is not already installed.
2. Run the storage emulator, instructions are available [here](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-emulator#start-and-initialize-the-storage-emulator).
    * Essentially run `AzureStorageEmulator.exe start` from the directory (or providing the path to the directory) where the exe lives.
3. Restart your function debugging.
4. You should no longer get the prompt when starting, nor should you see the runtime error above.

Since the audience of the article could well include groups that are not necessarily mutually-exclusive:

* Those new to Python
* Those new to Azure functions
* Those new to VS Code

It would be great if there was a note included in the documentation about this requirement for the Microsoft Azure storage emulator to be installed and running (the prompt from within VS Code doesn't provide a helpful enough message IMO).

The only reason I was able to fix this was because of my experience with functions being run from within Visual Studio where the storage emulator is automatically started, and I came across [this comment](https://github.com/Azure/azure-functions-core-tools/issues/1592#issuecomment-541362803) which helped.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 32211a5a-ad60-b982-b27e-1f2c7caef1c9
* Version Independent ID: 32a098cf-c39d-342b-5b7e-d9a62f6a9454
* Content: [Step 4: Debug the Azure Functions Python code locally with VS Code](https://docs.microsoft.com/en-us/azure/python/tutorial-vs-code-serverless-python-04?tabs=bash#feedback)
* Content Source: [python/tutorial-vs-code-serverless-python-04.md](https://github.com/MicrosoftDocs/azure-dev-docs/blob/master/python/tutorial-vs-code-serverless-python-04.md)
* Product: **azure-python**
* GitHub Login: @kraigb
* Microsoft Alias: **kraigb**

**Comment by kraigb** [comment]
@BeigeBadger Thanks. I'll get this incorporated into the docs when I have a chance.

**Comment by BeigeBadger** [comment]
Thanks for getting onto this so quickly @kraigb!

**Comment by kraigb** [comment]
Apologies for taking a long time to get to this, but I did update the article just now to include mention of the error. That said, in reviewing the whole process, the "UseDevelopmentStorage=true" value doesn't seem to be the default any longer...when I tested the process, the value for AzureWebJobsStorage was just an empty string. So the extension may have been updated in the last two months. In any case, thanks for pointing out the detail!
